03-rnaseq/clustering/clustering_heatmap_example.Rmd

```diff
@@ -0,0 +1,338 @@
+  dplyr::mutate(
+    groups = dplyr::case_when(
+    refinebio_treatment == "no treatment" ~ "control",
+    TRUE ~ "digoxin"
+    # Replace with group names relevant to your data
+    ),
+    # Create a variable to hold the sample summary information
+    # (excluding the lane information)
+    sample_info_summary = gsub("(_L).*", "", refinebio_title),
+    # Create a variable to store the mouse ids
+    mouse_id = stringr::word(refinebio_title, sep = "-|_", 2, 3),
+    # Create a variable to store the subgroup information
+    subgroup = dplyr::case_when(
+      startsWith(mouse_id, "2555MB") ~ "group 3",
+      startsWith(mouse_id, "1078MB") ~ "group 4",
+      TRUE ~ "unknown"
+  ))
+
+# Read in data tsv file
+df <- readr::read_tsv(file.path(
+  "data", # Replace with path to your data file
+  "SRP150101.tsv" # Replace with the name of your data file
+)) %>%
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>% dplyr::select(metadata$refinebio_accession_code)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$refinebio_accession_code)
+```
+
+## 3) Choose genes of interest
+Although you may want to create a heatmap including all of the genes in the set,
+alternatively, the heatmap could be created using only genes of interest. 
+For this example, we will sort genes by variance, but there are many alternative
+criterion by which you may want to sort your genes <i>eg</i> fold change,
+t-statistic, membership to a particular gene ontology, so on. 
+
+```{r Choose genes}
+# Calculate the variance for each gene
+variances <- apply(df, 1, var)
+
+# Determine the upper quartile variance cutoff value
+upper_var <- quantile(variances, 0.75)
+
+# Subset the data choosing only genes whose variances are in the upper quartile
+df_by_var <- df %>%
+  dplyr::filter(variances > upper_var)
+```
+
+## How are our technical replicates clustering together?
+
+### 4) Create a DESeqDataset
+A `DESeqDataSet` object is required to carry out the normalization step that
+you will see later using a function from the `DESeq2` package. This chunk of
+code will turn the data.frame (or matrix) into a `DESeqDataSet` object.
+
+```{r Create a DESeqDataset}
+# The `DESeqDataSetFromMatrix` function needs the values to be converted to
+# integers
+## TODO: Link to explanation about `DESeqDataSet` in the RNA-seq header section
+df_by_var <- df_by_var %>%
+  # Mutate numeric variables to be integers
+  dplyr::mutate_if(is.numeric, round)
+
+# Create a `DESeqDataSet` object
+dds <- DESeqDataSetFromMatrix(
+  countData = df_by_var,
+  colData = metadata,
+  design = ~ refinebio_treatment
+)
+```
+
+### 5) Prepare the metadata
+We are going to format the metadata to be used for annotation on the heatmap.
+
+```{r}
+# Let's prepare the metadata for the uncollapsed `DESeqData` set object
+metadata_uncollapsed <- metadata %>%
```

Comment:
This metadata will only be used for annotation correct? 
If yes, can we instead call this something like "annotation_df" or something like that? 

This is in part because I'd like to move away from using `collapsed` `uncollapsed` terminology and instead try to use terms that are more reflective of the samples and the question at hand. I think this will be helpful for our users to keep focused on the question/purpose of what we are doing.